### PR TITLE
Regression 

### PR DESCRIPTION
**Pull request by charlie-becker**
New:

1. Added refactored loss function for evidential regression `EvidentialRegLoss()`
2. Added new model class for evidential regression `EvidentialRegressorDNN_keras3()` that uses Keras subclassing
3. Added test for above model class

No current model classes or loss functions were replaced and only new ones were added. I would like thoughts on replacing the former model classes / loss functions with these in a separate PR, which will likely break other aspects of the code base such as the training scripts, notebooks, etc.

### CODE REVIEW

File: mlguess/keras/models.py [hunk 1]
```diff
@@ -1570,23 +1568,23 @@ class EvidentialRegressorDNN_keras3(keras.models.Model):
...
         use_noise=False,
         noise_sd=0.01,
-        lr=0.001,
+        lr=0.00001,
         use_dropout=False,
         dropout_alpha=0.1,
         batch_size=128,
         epochs=2,
-        kernel_reg="l2",
+        kernel_reg="l22345",
```
Comment (lucas-erlandson-nvidia) [inline mlguess/keras/models.py:1587]: Is this intentional? It seems that it will get converted to None later on

**Review by djgagne — APPROVED** [review]
It looks like @lucas-erlandson-nvidia 's comment was resolved. The other changes look good to me.
